web-local/src/lib/components/navigation/sources/sourceUtils.ts

```diff
@@ -32,3 +34,31 @@ export function compileCreateSourceSql(
     `)`
   );
 }
+
+export function inferSourceName(connector: V1Connector, path: string) {
```

Comment:
Can we have extensive tests for this? Also would the URL utility work for us instead of using split?

Reply:
Good call, just added tests.

[This URL utility](https://developer.mozilla.org/en-US/docs/Web/API/URL) doesn't come with a _slug_ attribute, so we'd need to use split regardless.